src/game_engine/unity/il2cpp.rs

```diff
@@ -50,74 +62,90 @@ impl Module {
             process.read::<Address32>(addr).ok()?.into()
         };
 
-        let type_info_definition_table_trg_addr = if is_64_bit {
+        let type_info_definition_table_trg_addr: Address = if is_64_bit {
             const TYPE_INFO_DEFINITION_TABLE_TRG_SIG: Signature<10> =
                 Signature::new("48 83 3C ?? 00 75 ?? 8B C? E8");
 
             let addr = TYPE_INFO_DEFINITION_TABLE_TRG_SIG
                 .scan_process_range(process, mono_module)?
                 .add_signed(-4);
 
-            addr + 0x4 + process.read::<i32>(addr).ok()?
+            process
+                .read::<Address64>(addr + 0x4 + process.read::<i32>(addr).ok()?)
+                .ok()?
+                .into()
         } else {
             const TYPE_INFO_DEFINITION_TABLE_TRG_SIG: Signature<10> =
                 Signature::new("C3 A1 ?? ?? ?? ?? 83 3C ?? 00");
 
             let addr =
                 TYPE_INFO_DEFINITION_TABLE_TRG_SIG.scan_process_range(process, mono_module)? + 2;
 
-            process.read::<Address32>(addr).ok()?.into()
+            process
+                .read::<Address32>(process.read::<Address32>(addr).ok()?)
+                .ok()?
+                .into()
         };
 
-        Some(Self {
-            is_64_bit,
-            version,
-            offsets: Offsets::new(version, is_64_bit)?,
-            assemblies: assemblies_trg_addr,
-            type_info_definition_table: type_info_definition_table_trg_addr,
+        if type_info_definition_table_trg_addr.is_null() {
+            None
+        } else {
+            Some(Self {
+                is_64_bit,
+                version,
+                offsets: Offsets::new(version, is_64_bit)?,
+                assemblies: assemblies_trg_addr,
+                type_info_definition_table: type_info_definition_table_trg_addr,
+            })
+        }
+    }
+
+    fn assemblies<'a>(
+        &'a self,
+        process: &'a Process,
+    ) -> impl DoubleEndedIterator<Item = Assembly> + 'a {
+        let (assemblies, nr_of_assemblies): (Address, u64) = if self.is_64_bit {
+            let [first, limit] = process
+                .read::<[u64; 2]>(self.assemblies)
+                .unwrap_or_default();
+            let count = (limit - first) / self.size_of_ptr();
```

Comment:
This (and below) shouldn't be a normal subtraction, as Rust panics on overflow in debug mode here. Considering we may read garbage, it would be unfortunate for the auto splitter to panic. wrapping_sub or so should probably be used.

Reply:
If we read garbage data, then it would make more sense to use `saturating_sub()` so we still stay at the bound of the type instead of underflowing. It also means `count` will become `0`, returning an empty iterator at the end of the function